src/ui/views/tui_folder_view.rs

```diff
@@ -45,25 +41,11 @@ impl<'a> Widget for TuiFolderView<'a> {
         let config = self.context.config_ref();
         let display_options = config.display_options_ref();
 
-        let (default_layout, constraints): (bool, &[Constraint; 3]) =
-            if !display_options.collapse_preview() {
-                (true, &display_options.default_layout)
-            } else {
-                match child_list {
-                    Some(_) => (true, &display_options.default_layout),
-                    None => match curr_entry {
-                        None => (false, &display_options.no_preview_layout),
-                        Some(e) => match preview_context.get_preview_ref(e.file_path()) {
-                            Some(Some(p)) if p.status.code() != Some(1) => {
-                                (true, &display_options.default_layout)
-                            }
-                            _ => (false, &display_options.no_preview_layout),
-                        },
-                    },
-                }
-            };
+        let constraints = get_constraints(self.context);
+        let is_default_layout = constraints == &display_options.default_layout;
 
-        let layout_rect = if config.display_options_ref().show_borders() {
+        let layout_rect = if display_options.show_borders() {
+            let layout = calculate_layout_with_borders(area, constraints);
```

Comment:
We do rundandant calculations because we kind of have to render while figuring out the eventual layout